packages/rrweb/src/utils.ts

```diff
@@ -203,14 +205,17 @@ export function isBlocked(
       ? (node as HTMLElement)
       : node.parentElement;
   if (!el) return false;
-
+  
   if (typeof blockClass === 'string') {
     if (el.classList.contains(blockClass)) return true;
     if (checkAncestors && el.closest('.' + blockClass) !== null) return true;
   } else {
     if (classMatchesRegex(el, blockClass, checkAncestors)) return true;
   }
-  return false;
+  if (blockSelector) {
+    if ((node as HTMLElement).matches(blockSelector)) return true;
+  }
```

Comment:
If `checkAncestors` is true we should also check to see if any of the parents of this node are blocked

```suggestion
  if (blockSelector) {
    if ((node as HTMLElement).matches(blockSelector)) return true;
    if (checkAncestors && el.closest(blockSelector) !== null) return true;
  }
```